yb-voyager/src/migassessment/gather.go

```diff
@@ -171,22 +170,69 @@ func (pt *progressTracker) printSingleLine(progress NodeProgress) {
 	fmt.Printf("\r\033[K  %s %-*s %s\n", statusIcon, pt.maxNameLen+1, displayName+":", progress.Stage)
 }
 
-// GatherAssessmentMetadataFromPG collects metadata from PostgreSQL primary and replicas.
-// Accepts the validated replicas to collect from, and returns the list of failed replica names
-// (for reporting partial multi-node assessment).
-// Collection is performed in parallel for better performance.
+// finalizeAndCheckResults processes the final results from all collection goroutines.
+// It logs each result and returns an error if any node failed.
+func (pt *progressTracker) finalizeAndCheckResults(resultChan <-chan collectionResult) error {
+	var failedNodes []string
+	for result := range resultChan {
+		displayName := pt.displayNames[result.nodeName]
+		if result.err == nil {
+			log.Infof("Successfully collected metadata from %s", displayName)
+		} else {
+			log.Errorf("Metadata collection failed on %s: %v", displayName, result.err)
```

Comment:
so, right now, if metadata collection fails for some users, 
- it is just logged 
- on console, we simply print : 'metadata collection failed on one or more nodes" ?

This isn't ideal, because then the user has to take an extra step of looking through the logs to figure out what happened. 

Can we wrap the individual errors into the error we are returning, and print it properly? 

Let's treat this as an enhancement, don't want to block this PR further, as it's a big one.